feat: use endpoints compatible with LFS server discovery

vit-zikmund opened the pull request:
Introducing the base URI via the `BaseView` seems like a nice and consistent way of controlling this.

This change additionally adds support for the `<organization>` placeholder in the URI to contain slashes, as some git repositories (e.g. those GitLab) contain multiple levels of organization groups/namespaces.

As it currently stands in the PR, the whole such string ends up in the `organization` variable, which I feel, became a bit misleading. I'd recommend refactoring the code to use something like `org_path`, but wanted to hear your opinions.

Closes: https://github.com/datopian/giftless/issues/151

vit-zikmund commented:
OK, that's more than fair. Feature flag incoming.

athornton reviewed (approved):
This looks awesome.

athornton commented:
If no one from Datopian objects I'll merge it tomorrow.

vit-zikmund commented:
Please don't merge yet, just discovered Werkzeug kinda fails at parsing the `/<repo>.git/` part, where the `repo` variable actually gets the full string between slashes. I'll have to work around it. But it's too early today for me to do anything more on this, gotta chase the missed yesterday's sleep.

athornton commented:
Cool.  You will want to rebase, as I merged a couple of smaller PRs (one of them yours) before getting here.

vit-zikmund commented:
I should be done here. I've added the handful of promised tests for the legacy endpoints and discovered, that the previous error I was seeing (and blamed to Werkzeug) is actually my bug, I fixed on the [other PR](https://github.com/datopian/giftless/pull/149/commits/d622b9104d316ff4bdd52a0b99b54b9092d41ee2) :)
